### PR TITLE
CMake: the Reformatting

### PR DESCRIPTION
**Pull request by marcusmueller**
# Pull Request Details

We have automated formatting, enforced for C++. We have PEP8 for Python.

Why can't we have some assistance with the least beautiful language in tree?

Enter [cmake-format](https://cmake-format.readthedocs.io/en/latest/), which does exactly that. Only needed to write a short config file (actually, a Python module), a minimal amount of glue to bind a few settings in that to what's in .clang-format, and then run

```
cmake-format -c .cmake-format.py \
             -i \
             cmake/cmake_uninstall.cmake.in cmake/Modules/* \
             cmake/Toolchains/arm_cortex_a8_native.cmake cmake/Toolchains/oe-sdk_cross.cmake 
             $(git ls-files | grep CMakeLists.txt )
```

## Which blocks/areas does this affect?

CMake.

## Testing Done

Still builds like a champ!

I don't want to enforce running this on every commit – I have no idea how
stable cmakelang is medium-term. We could add this to a pre-commit hook,
together with flake8 and clang-formatting, though.

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/master/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/master/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/master/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [x] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Comment by mormj** [comment]
Are you suggesting this for 3.10 or start this with 3.11?

**Comment by willcode** [comment]
It this all whitespace mods, or does it do re-ordering and other changes?

**Comment by marcusmueller** [comment]
> Are you suggesting this for 3.10 or start this with 3.11?

@mormj I'd be suggesting this for 3.11; not a functional change, though.

> It this all whitespace mods, or does it do re-ordering and other changes?

Whitespace, indents, line breaks, consistent capitalizations (`builtin_function(KEYWORD OthEr_stuff)`); no reordering. It could have reordered argument lists where that's permitted, but we have no instance of that, far as I could find in the tree.

**Comment by marcusmueller** [comment]
Noticed that setting the indent depth to 2 instead of 4 would possibly yield a smaller diff. Must investigate.

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,10 +44,10 @@ message(STATUS "Build date set to ${BUILD_DATE}.")
 include(GrComponent)
 
 # Set the version information here
-SET(VERSION_MAJOR 3)
-SET(VERSION_API   10)
-SET(VERSION_ABI   0)
-SET(VERSION_PATCH 0-rc2)
+set(VERSION_MAJOR 3)
+set(VERSION_API 10)
```
Comment (mbr0wn) [inline CMakeLists.txt:48]: I actually preferred the old indenting here.

**Review by mbr0wn — APPROVED** [review]
Much nicer.

File: CMakeLists.txt (hunk 1 above)
Comment (marcusmueller) [reply CMakeLists.txt:48]: agreed, don't know whether cmake-lang supports that, though. Haven't found an option, so [asking on SO](https://stackoverflow.com/questions/70597446/keep-multiple-space-aligned-setvariable-value-the-way-they-were-during-cmake).

**Comment by mormj** [comment]
Holding off a bit more on this until we are done with emergency patch releases for 3.10.  Maybe should be merged closer to the 3.10.1.0 window

File: CMakeLists.txt (hunk 1 above)
Comment (mbr0wn) [reply CMakeLists.txt:48]: Isn't this all you need? https://cmake-format.readthedocs.io/en/stable/format-features.html?highlight=disable#disable-formatting-locally

Comment (marcusmueller) [reply CMakeLists.txt:48]: Yep! As [someone helpfully pointed out as well](https://stackoverflow.com/questions/70597446/keep-multiple-space-aligned-setvariable-value-the-way-they-were-during-cmake/70846835#70846835) :)

**Comment by willcode** [comment]
Close in favor of #6100?

**Comment by mormj** [comment]
> Close in favor of #6100?

Yes.  6100 has the main commit, then the aftermath of running the script again on the latest `main`

**Pull request closed without merging**